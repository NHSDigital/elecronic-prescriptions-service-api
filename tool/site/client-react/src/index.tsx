import {Bundle} from "fhir/r4"
import * as React from "react"
import {
  createPrescriptionSummaryProps,
  PrescriptionSummary
} from "./components/prescription-summary/prescriptionSummary"
import {PageContainer} from "./components/pageContainer"
import ReactDOM = require("react-dom")

(async function() {
  const baseUrl = "/"
  const urlParams = new URLSearchParams(window.location.search)
  const prescriptionId = urlParams.get("prescription_id")
  const getPrescriptionResponse: Response = await fetch(`${baseUrl}prescription/${prescriptionId}`)
  const bundle: Bundle = await getPrescriptionResponse.json()
  const prescriptionSummaryProps = createPrescriptionSummaryProps(bundle)
  const content = (
    <PageContainer>
      <PrescriptionSummary
        patient={prescriptionSummaryProps.patient}
        practitionerRole={prescriptionSummaryProps.practitionerRole}
      />
    </PageContainer>
  )
  ReactDOM.render(content, document.getElementById("root"))
}())
