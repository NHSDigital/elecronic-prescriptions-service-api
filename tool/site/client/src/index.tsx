import * as React from "react"
import {Page} from "./components/page"
import SendPreSignPage from "./pages/sendPreSignPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPostSignPage from "./pages/sendPostSignPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import VerifyPage from "./pages/verifyPage"
import ComparePage from "./pages/comparePage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SendPreSignPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPostSignPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/verify`}>
                <PageContainer>
                  <VerifyPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
