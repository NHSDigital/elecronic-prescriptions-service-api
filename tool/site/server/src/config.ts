import {Environment} from "./services/environment"

export interface Config {
  commitId: string
  validatorVersion: string
  basePath: string
  baseUrl: string
  environment: Environment
  sessionKey: string
  privateApigeeUrl: string
  publicApigeeUrl: string
  clientId: string
  clientSecret: string
  privateKey: string
  keyId: string
  subject: string
  cis2Secret: string
  cis2EgressHost: string
  refreshTokenTimeout: number
}

export const CONFIG: Config = {
  commitId: process.env.COMMIT_ID ?? "unknown",
  validatorVersion: process.env.VALIDATOR_VERSION ?? "unknown",
  basePath: process.env.BASE_PATH ?? "eps-api-tool",
  baseUrl: process.env.BASE_PATH ? `/${process.env.BASE_PATH}/` : "/",
  environment: process.env.ENVIRONMENT as Environment ?? "int",
  sessionKey: process.env.SESSION_TOKEN_ENCRYPTION_KEY ?? "",
  privateApigeeUrl: `https://${process.env.APIGEE_DOMAIN_NAME}`,
  publicApigeeUrl: process.env.PUBLIC_APIGEE_URL ?? "",
  clientId: process.env.DEMO_APP_CLIENT_ID ?? "",
  clientSecret: process.env.DEMO_APP_CLIENT_KEY ?? "",
  privateKey: process.env.DEMO_APP_PRIVATE_KEY ?? "",
  keyId: process.env.DEMO_APP_KEY_ID ?? "",
  subject: process.env.APP_JWT_SUBJECT ?? "",
  cis2Secret: process.env.DEMO_APP_CIS2_KEY ?? "",
  cis2EgressHost: process.env.CIS2_EGRESS_HOST ?? "",
  // full refresh timeout seconds is divided by 3 as only one refresh is working atm
  // times by 1000 to get miliseconds as ttls for cookies takes ms
  refreshTokenTimeout: (3599 / 3) * 1000
}
