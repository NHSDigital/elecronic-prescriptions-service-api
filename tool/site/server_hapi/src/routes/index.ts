import statusRoutes from "./health/get-status"
import accessTokenRoutes from "./auth/login"
import editRoutes from "./prescribe/edit"
import signRoutes from "./prescribe/sign"
import sendRoutes from "./prescribe/send"

const authRoutes = [
  ...accessTokenRoutes
]

const prescribeRoutes = [
  ...editRoutes,
  ...signRoutes,
  ...sendRoutes
]

const healthcheckRoutes = [
  ...statusRoutes
]

const routes = [
  ...authRoutes,
  ...healthcheckRoutes,
  ...prescribeRoutes
]

export default routes
