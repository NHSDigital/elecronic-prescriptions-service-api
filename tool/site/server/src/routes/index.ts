import statusRoutes from "./health/get-status"
import loginRoute from "./auth/login"
import oauthCallbackRoute from "./auth/callback"
import refreshRoute from "./auth/refresh"
import logoutRoute from "./auth/logout"
import configRoutes from "./config/config"
import sessionRoutes from "./state/session"
import editRoutes from "./prescribe/edit"
import signRoutes from "./prescribe/sign"
import sendRoutes from "./prescribe/send"
import cancelRoutes from "./prescribe/cancel"
import validatorRoutes from "./validate/validator"
import searchRoutes from "./tracker/tracker"
import releaseRoutes from "./dispense/release"
import verifyRoutes from "./dispense/verify"
import returnRoutes from "./dispense/return"
import dispenseRoutes from "./dispense/dispense"
import claimRoutes from "./dispense/claim"
import withdrawRoutes from "./dispense/withdraw"
import comparePrescriptions from "./api/comparePrescriptions"
import {isSandbox} from "../services/environment"
import {CONFIG} from "../config"

const authRoutes = [
  loginRoute,
  oauthCallbackRoute,
  refreshRoute,
  logoutRoute
]

const apiRoutes = [
  ...comparePrescriptions
]

const stateRoutes = [
  ...sessionRoutes
]

const prescribingRoutes = [
  ...editRoutes,
  ...signRoutes,
  ...sendRoutes,
  ...cancelRoutes
]

const validateRoutes = [
  ...validatorRoutes
]

const dispensingRoutes = [
  ...releaseRoutes,
  ...verifyRoutes,
  ...returnRoutes,
  ...dispenseRoutes,
  ...claimRoutes,
  ...withdrawRoutes
]

const trackerRoutes = [
  ...searchRoutes
]

const healthcheckRoutes = [
  ...statusRoutes
]

const routes = isSandbox(CONFIG.environment)
  ? []
  : [
    configRoutes,
    ...authRoutes,
    ...apiRoutes,
    ...stateRoutes,
    ...healthcheckRoutes,
    ...prescribingRoutes,
    ...validateRoutes,
    ...dispensingRoutes,
    ...trackerRoutes
  ]

export default routes
