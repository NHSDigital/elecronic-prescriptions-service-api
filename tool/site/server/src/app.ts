import Hapi from "@hapi/hapi"
import routes from "./routes"
import HapiPino from "hapi-pino"
import Vision from "@hapi/vision"
import * as inert from "@hapi/inert"
import Yar from "@hapi/yar"
import Cookie from "@hapi/cookie"
import {isDev, isLocal} from "./services/environment"
import axios from "axios"
import {setSessionValue} from "./services/session"

const init = async () => {
  axios.defaults.validateStatus = () => true

  const server = Hapi.server({
    port: 9000,
    host: "0.0.0.0",
    routes: {
      cors: true // Won't run as Apigee hosted target without this
    }
  })

  const baseUrl = process.env.BASE_PATH
    ? `/${process.env.BASE_PATH}/`
    : "/"

  await server.register(Cookie)
  server.auth.strategy("session", "cookie", {
    cookie: {
      name: "auth",
      password: process.env.SESSION_TOKEN_ENCRYPTION_KEY,
      isSecure: true
    },
    redirectTo: (request: Hapi.Request) => {
      if (isDev()) {
        setSessionValue(
          "use_signing_mock",
          request.query["use_signing_mock"],
          request
        )
      }
      return `${baseUrl}login`
    }
  })
  server.auth.default("session")

  server.route(routes)

  await server.register({
    plugin: Yar,
    options: {
      storeBlank: false,
      // Use "0" maxCookieSize to force all session data to be written to cache
      maxCookieSize: 0,
      cache: {
        expiresIn: 24 * 60 * 60 * 1000
      },
      cookieOptions: {
        password: process.env.SESSION_TOKEN_ENCRYPTION_KEY ?? "",
        isSecure: true,
        isSameSite: "None"
      }
    }
  })

  await server.register({
    plugin: HapiPino,
    options: {
      // Pretty print in local environment only to avoid spamming logs
      prettyPrint: isLocal(),
      // Redact Authorization headers, see https://getpino.io/#/docs/redaction
      redact: ["req.headers.authorization"]
    }
  })

  await server.register(inert)

  server.route({
    method: "GET",
    path: `/static/{param*}`,
    options: {
      auth: false
    },
    handler: {
      directory: {
        path: "static"
      }
    }
  })

  await server.register(Vision)

  server.views({
    engines: {
      html: require("handlebars")
    },
    relativeTo: __dirname,
    path: "templates"
  })

  server.route(addHomeViewRoute())
  server.route(addViewRoute("login", true))
  server.route(addViewRoute("my-prescriptions"))
  server.route(addViewRoute("validate"))
  server.route(addViewRoute("search"))
  server.route(addViewRoute("prescribe/load"))
  server.route(addViewRoute("prescribe/edit"))
  server.route(addViewRoute("prescribe/send"))
  server.route(addViewRoute("prescribe/cancel"))
  server.route(addViewRoute("dispense/release"))
  server.route(addViewRoute("dispense/return"))
  server.route(addViewRoute("dispense/dispense"))
  server.route(addViewRoute("dispense/withdraw"))
  server.route(addViewRoute("dispense/claim"))

  function addHomeViewRoute() : Hapi.ServerRoute {

    const baseUrl = process.env.BASE_PATH
      ? `/${process.env.BASE_PATH}/`
      : "/"

    return {
      method: "GET",
      path: `/`,
      handler: {
        view: {
          template: "index",
          context: {
            baseUrl,
            environment: process.env.ENVIRONMENT
          }
        }
      }
    }
  }

  function addViewRoute(path: string, skipAuth?: boolean): Hapi.ServerRoute {
    const baseUrl = process.env.BASE_PATH
      ? `/${process.env.BASE_PATH}/`
      : "/"

    const viewRoute = {
      method: "GET",
      path: `/${path}`,
      handler: {
        view: {
          template: "index",
          context: {
            baseUrl,
            environment: process.env.ENVIRONMENT
          }
        }
      }
    }

    if (skipAuth) {
      return {
        ...viewRoute,
        options: {
          auth: false
        }
      }
    }

    return viewRoute
  }

  await server.start()
  server.log("info", `Server running on ${server.info.uri}`)
}

process.on("unhandledRejection", err => {
  console.log(err)
  process.exit(1)
})

init()
