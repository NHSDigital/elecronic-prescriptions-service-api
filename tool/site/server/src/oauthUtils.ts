import ClientOAuth2 from "client-oauth2"
import {URL} from "url"
import {CONFIG} from "./config"
import {getRegisteredCallbackUrl} from "./routes/helpers"

export interface OAuthClient {
  getUri: (options?: ClientOAuth2.Options) => string
  getToken: (uri: URL) => Promise<Token>
}

export interface Token {
  data: ClientOAuth2.Data
  tokenType: string
  accessToken: string
  refreshToken: string
}

const oauthClient =
  new ClientOAuth2({
    clientId: CONFIG.clientId,
    clientSecret: CONFIG.clientSecret,
    redirectUri: getRegisteredCallbackUrl("callback"),
    accessTokenUri: `${CONFIG.privateApigeeUrl}/oauth2/token`,
    authorizationUri: `${CONFIG.publicApigeeUrl}/oauth2/authorize`,
    body: {
      client_id: CONFIG.clientId,
      client_secret: CONFIG.clientSecret
    }
  })

export default function getOAuthClient(): OAuthClient {
  return oauthClient.code
}

export async function refreshToken(data: ClientOAuth2.Data): Promise<ClientOAuth2.Token> {
  const oauthClientToken = oauthClient.createToken(data)
  const refreshedToken = await oauthClientToken.refresh()
  oauthClientToken.expiresIn(parseInt(oauthClientToken.data.refreshTokenExpiresIn))
  return refreshedToken
}
