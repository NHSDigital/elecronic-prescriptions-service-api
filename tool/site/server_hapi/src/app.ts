import Hapi from "@hapi/hapi"
import routes from "./routes"
import HapiPino from "hapi-pino"

const init = async () => {
  const server = Hapi.server({
    port: 9002,
    host: "0.0.0.0",
    routes: {
      cors: true, // Won't run as Apigee hosted target without this
      payload: {
        parse: false
      }
    }
  })

  server.route(routes)

  await server.register({
    plugin: HapiPino,
    options: {
      // Dont pretty print to avoid spamming logs
      prettyPrint: false,
      // Redact Authorization headers, see https://getpino.io/#/docs/redaction
      redact: ["req.headers.authorization"]
    }
  })

  await server.start()
  server.log("info", `Server running on ${server.info.uri}`)
}

process.on("unhandledRejection", (err) => {
  console.log(err)
  process.exit(1)
})

init()
