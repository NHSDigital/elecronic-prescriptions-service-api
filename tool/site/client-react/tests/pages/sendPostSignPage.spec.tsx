import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import SendPostSignPage from "../../src/pages/sendPostSignPage"

const baseUrl = "baseUrl/"
const token = "MzQxMWJmMjUtMDNlMy00N2FiLWEyOGItMGIyYjVlNTg4ZGU3"
const prescriptionId = "003D4D-A99968-4C5AAJ"
const context: AppContextValue = {baseUrl}

const sendUrl = `${baseUrl}prescribe/send`

beforeEach(() => moxios.install())

afterEach(() => moxios.uninstall())

test("Displays loading text while prescription is being sent", async () => {
  const {container} = renderWithContext(<SendPostSignPage token={token}/>, context)
  await waitFor(() => screen.getByText("Sending prescription(s)."))

  expect(screen.getByText("Loading...")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays confirmation page if single prescription is sent successfully", async () => {
  moxios.stubRequest(sendUrl, {
    status: 200,
    response: {
      success: true,
      prescription_id: prescriptionId,
      request: "JSON Request",
      request_xml: "XML Request",
      response: "JSON Response",
      response_xml: "XML Response"
    }
  })

  const container = await renderPage()

  expect(screen.getByText(prescriptionId)).toBeTruthy()
  expect(screen.getByText(JSON.stringify("JSON Request"))).toBeTruthy()
  expect(screen.getByText("XML Request")).toBeTruthy()
  expect(screen.getByText(JSON.stringify("JSON Response"))).toBeTruthy()
  expect(screen.getByText("XML Response")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays confirmation page if multiple prescriptions are sent successfully", async () => {
  moxios.stubRequest(sendUrl, {
    status: 200,
    response: {
      results: [
        {
          prescription_id: "003D4D-A99968-4C5AAJ",
          success: true
        },
        {
          prescription_id: "008070-A99968-41CD9V",
          success: true
        },
        {
          prescription_id: "010E34-A99968-467D9Z",
          success: false
        }
      ]
    }
  })

  const container = await renderPage()

  expect(screen.getByText("003D4D-A99968-4C5AAJ")).toBeTruthy()
  expect(screen.getByText("008070-A99968-41CD9V")).toBeTruthy()
  expect(screen.getByText("010E34-A99968-467D9Z")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

async function renderPage() {
  const {container} = renderWithContext(<SendPostSignPage token={token}/>, context)
  await waitFor(() => screen.getByText(/Send Result/))
  return container
}
