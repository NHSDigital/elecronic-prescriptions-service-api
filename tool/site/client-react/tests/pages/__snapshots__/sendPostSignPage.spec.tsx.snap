// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays confirmation page if multiple prescriptions are sent successfully 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Send Results</label></h1>
<div class=\\"sc-bdvvtL jjZBkx\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy Prescription IDs</button></div>
<div class=\\"nhsuk-table-responsive\\">
  <table class=\\"nhsuk-table\\">
    <thead class=\\"nhsuk-table__head\\">
      <tr class=\\"nhsuk-table__row\\">
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">ID</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Success</th>
      </tr>
    </thead>
    <tbody class=\\"nhsuk-table__body\\">
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">003D4D-A99968-4C5AAJ</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
          </svg></td>
      </tr>
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">008070-A99968-41CD9V</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
          </svg></td>
      </tr>
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">010E34-A99968-467D9Z</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__cross\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path d=\\"M17 18.5c-.4 0-.8-.1-1.1-.4l-10-10c-.6-.6-.6-1.6 0-2.1.6-.6 1.5-.6 2.1 0l10 10c.6.6.6 1.5 0 2.1-.3.3-.6.4-1 .4z\\"></path>
            <path d=\\"M7 18.5c-.4 0-.8-.1-1.1-.4-.6-.6-.6-1.5 0-2.1l10-10c.6-.6 1.5-.6 2.1 0 .6.6.6 1.5 0 2.1l-10 10c-.3.3-.6.4-1 .4z\\"></path>
          </svg></td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`Displays confirmation page if single prescription is sent successfully 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Send Result <svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
    </svg></label></h1>
<dl class=\\"nhsuk-summary-list\\">
  <div class=\\"nhsuk-summary-list__row\\">
    <dt class=\\"nhsuk-summary-list__key\\">ID</dt>
    <dd class=\\"nhsuk-summary-list__value\\">003D4D-A99968-4C5AAJ</dd>
  </div>
</dl>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/prescribe/cancel?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">Cancel prescription</span></a></div>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/release?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">Release prescription</span></a></div>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/dispense?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">Dispense prescription</span></a></div>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/claim?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">Claim for prescription</span></a></div>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/search?prescription_id=003D4D-A99968-4C5AAJ\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">View prescription</span></a></div>
<div class=\\"nhsuk-expander-group\\">
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Request (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div class=\\"sc-bdvvtL jjZBkx\\"><button class=\\"nhsuk-button sc-dkPtRN hhyeAF\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-dkPtRN hhyeAF\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22JSON%20Request%22\\" download=\\"message\\">Download</a></div><pre class=\\"sc-gsDKAQ jsqwND\\">\\"JSON Request\\"</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Request (HL7 V3)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div class=\\"sc-bdvvtL jjZBkx\\"><button class=\\"nhsuk-button sc-dkPtRN hhyeAF\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-dkPtRN hhyeAF\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:text/xml;charset=utf-8,XML%20Request\\" download=\\"message\\">Download</a></div><pre class=\\"sc-gsDKAQ jsqwND\\">XML Request</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Response (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div class=\\"sc-bdvvtL jjZBkx\\"><button class=\\"nhsuk-button sc-dkPtRN hhyeAF\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-dkPtRN hhyeAF\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22JSON%20Response%22\\" download=\\"message\\">Download</a></div><pre class=\\"sc-gsDKAQ jsqwND\\">\\"JSON Response\\"</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Response (HL7 V3)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div class=\\"sc-bdvvtL jjZBkx\\"><button class=\\"nhsuk-button sc-dkPtRN hhyeAF\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-dkPtRN hhyeAF\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:text/xml;charset=utf-8,XML%20Response\\" download=\\"message\\">Download</a></div><pre class=\\"sc-gsDKAQ jsqwND\\">XML Response</pre>
    </div>
  </details>
</div>
<div class=\\"sc-bdvvtL jjZBkx\\"><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"button\\">Back</button></div>"
`;

exports[`Displays loading text while prescription is being sent 1`] = `"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Loading...</label></h1><label class=\\"nhsuk-label\\">Sending prescription(s).</label>"`;
